A3A/addons/core/functions/Missions/fn_RIV_ATT_Transfer.sqf

```diff
@@ -0,0 +1,531 @@
+private _route = [_posOrigin, _posDest] call A3A_fnc_findPath;
+private _pathState = [];
+
+_route = _route apply { _x select 0 };			// reduce to position array
+if (_route isEqualTo []) then { _route = [_posOrigin, _posDest] };
+
+//private _route = [_transferConvoySpawnPos, _hideoutPosition] call A3A_fnc_findPath;
+diag_log _hideoutPosition;
+diag_log _hideoutPosition;
+diag_log _hideoutPosition;
+diag_log _hideoutPosition;
+diag_log _hideoutPosition;
+diag_log _route;
+diag_log _route;
+diag_log _route;
+diag_log _route;
+diag_log _route;
+diag_log _route;
+//////////////////////////////////////////////
+//  Task        	                        //
+//////////////////////////////////////////////
+private _taskId = "RIV_ATT" + str A3A_taskCount;
+
+[
+    [teamPlayer,civilian],
+    _taskId,
+    [
+        format [localize "STR_RIV_ATT_transfer_text", A3A_faction_riv get "name", ([_marker] call A3A_fnc_localizar), _displayTime],
+        format [localize "STR_RIV_ATT_transfer_header", A3A_faction_riv get "name"],
+        _marker
+    ],
+    _hideoutPosition,
+    false,
+    0,
+    true,
+    "destroy",///maybe change the icon
+    true
+] call BIS_fnc_taskCreate;
+[_taskId, "RIV_ATT", "CREATED"] remoteExecCall ["A3A_fnc_taskUpdate", 2];
+
+waitUntil {
+    sleep 1;
+    (call SCRT_fnc_misc_getRebelPlayers) inAreaArray [_hideoutPosition, distanceSPWN1, distanceSPWN1] isNotEqualTo [] || {dateToNumber date > _dateLimitNum}
+};
+
+private _lootContainer = nil;
+private _vehObj = nil;
+
+if (dateToNumber date < _dateLimitNum) then {
+    private _tempVeh = "Land_LampShabby_off_F" createVehicleLocal _hideoutPosition;
+    private _atlPos = getPosATL _tempVeh;
+    deleteVehicle _tempVeh;
+
+    private _compositionMap = createHashMapFromArray [
+        ["COMP1", SCRT_fnc_composition_rivals1],
+        ["COMP2", SCRT_fnc_composition_rivals2],
+        ["COMP3", SCRT_fnc_composition_rivals3]
+    ];
+
+    private _fnc = _compositionMap get (selectRandom ["COMP1", "COMP2", "COMP3"]);
+
+    private _objects = [_atlPos, (random 360), (call _fnc)] call BIS_fnc_objectsMapper;
+    {_x setVectorUp surfaceNormal getPos _x} forEach _objects;
+
+    _vehicles append _objects;
+
+    //////////////////////////////////////////////
+    //  Loot or Vehicle	with loot               //
+    //////////////////////////////////////////////
+    private _iterations = 0;
+    private _lootContainerPosition = nil;
+
+    while {true} do {
+        _lootContainerPosition = [
+            _hideoutPosition, //center
+            0, //minimal distance
+            25, //maximumDistance
+            3, //object distance
+            0, //water mode
+            0.45, //maximum terrain gradient
+            0, //shore mode
+            [], //blacklist positions
+            [_hideoutPosition, _hideoutPosition] //default position
+        ] call BIS_fnc_findSafePos;
+
+        if (_iterations isEqualTo 50) exitWith {};
+        _iterations = _iterations + 1;
+    };
```

Comment:
Didn't we discuss alternate methods that don't require using while loops?

Reply:
I copied it from Hideout mission, soooo....
it was there before me

Reply:
Sounds about right